quantumblacklabs/kedro-viz
# [KED-1657] Seed randomly-generated data with UUIDs

## Description

1. Generate a random UUID, and log it to the console, if random data is selected. Use it to seed all the random number generators.
2. Allow users to copy the ID from the console and add it to a query string in the URL, so that the randomly-generated data can be replicated.

## Development notes

1. This uses https://github.com/davidbau/seedrandom, and globally modifies the behaviour of `Math.random`. Since it can only be called when the app is being run in standalone mode, it shouldn't have any untoward effects, as far as I can tell.
2. Uses the URL.searchParams web API, which is not supported in IE.

## QA notes

Check the console when on `?data=random`, and try clicking the URL to keep the data as is.
Example output:
![image](https://user-images.githubusercontent.com/1155816/83050953-37f1c300-a045-11ea-9210-14c512d8f188.png)

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
